main.tf

```diff
@@ -56,7 +56,7 @@ resource "ibm_database" "mongodb" {
   tags                      = var.tags
   service_endpoints         = var.endpoints
   plan_validation           = var.plan_validation
-  configuration             = var.configuration != null ? jsonencode(var.configuration) : null
+  configuration             = var.configuration != null ? jsonencode({ for k, v in var.configuration : k => v if v != null }) : null
```

Comment:
maybe leave a code comment above this to say why we need to do this. Perhaps add the root cause into https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/issues/273 and just link to that issue?